src/lsp_data.rs

```diff
@@ -87,72 +90,56 @@ impl Location {
-                    line: line,
-                    character: col,
+                    line: from_usize(line),
+                    character: from_usize(col),
                 },
             },
         }
     }
 }
 
-/// The initialize request is sent as the first request from the client to the server.
-#[allow(non_snake_case)]
-#[derive(Debug, Deserialize)]
-pub struct InitializeParams {
-    pub processId: usize,
-    pub rootPath: PathBuf
-}
-
-#[derive(Debug, Deserialize)]
-pub struct Document {
-    pub uri: String
+pub fn sk_from_def_kind(k: raw::DefKind) -> SymbolKind {
+    match k {
+        raw::DefKind::Enum => SymbolKind::Enum,
+        raw::DefKind::Tuple => SymbolKind::Array,
+        raw::DefKind::Struct => SymbolKind::Class,
+        raw::DefKind::Trait => SymbolKind::Interface,
+        raw::DefKind::Function => SymbolKind::Function,
+        raw::DefKind::Method => SymbolKind::Function,
+        raw::DefKind::Macro => SymbolKind::Function,
+        raw::DefKind::Mod => SymbolKind::Module,
+        raw::DefKind::Type => SymbolKind::Interface,
+        raw::DefKind::Local => SymbolKind::Variable,
+        raw::DefKind::Static => SymbolKind::Variable,
+        raw::DefKind::Const => SymbolKind::Variable,
+        raw::DefKind::Field => SymbolKind::Variable,
+        raw::DefKind::Import => SymbolKind::Module,
+    }
 }
 
-impl_file_name!(Document);
-
-#[allow(non_snake_case)]
-#[derive(Debug, Deserialize)]
-pub struct TextDocumentIdentifier {
-    pub uri: String
+pub fn new_completion_item(label: String, detail: String) -> CompletionItem {
+	CompletionItem {
+        label : label,
+        kind: None,
+        detail: Some(detail),
+        documentation: None,
+        sort_text: None,
+        filter_text: None,
+        insert_text: None,
+        text_edit: None,
+        additional_text_edits: None,
+        command: None,
+        data: None,
+	}
 }
 
-impl_file_name!(TextDocumentIdentifier);
-
-#[allow(non_snake_case)]
-#[derive(Debug, Deserialize)]
-pub struct VersionedTextDocumentIdentifier {
-    pub version: u64,
-    pub uri: String
-}
+pub type HoverParams = TextDocumentPositionParams;
 
-/// An event describing a change to a text document. If range and rangeLength are omitted
-/// the new text is considered to be the full content of the document.
-#[allow(non_snake_case)]
-#[derive(Debug, Deserialize)]
-pub struct TextDocumentContentChangeEvent {
-    pub range: Range,
-    pub rangeLength: Option<u32>,
-    pub text: String
-}
-
-#[allow(non_snake_case)]
-#[derive(Debug, Deserialize)]
-pub struct ReferenceContext {
-    pub includeDeclaration: bool,
-}
-
-/// Represents information about programming constructs like variables, classes,
-/// interfaces etc.
-#[derive(Debug, Serialize)]
-pub struct SymbolInformation {
-    pub name: String,
-    pub kind: u32, // can be made to numeric enum
-    pub location: Location,
-}
+/* ----------------- These are not LSP types: ----------------- */
```

Comment:
The comment is a reflection that those types should probably be moved elsewhere (I guess it's more of a TODO). But I don't know where.